.github/workflows/system-tests.yml

```diff
@@ -216,7 +211,7 @@ jobs:
         run: |
           docker image list
       - name: Run scenario
-        run: ./run.sh ${{ matrix.scenario }}
+        run: ./build.sh --images runner && ./run.sh ${{ matrix.scenario }}
```

Comment:
In a few other repos I see the build step being separate from the run step. Should we do that here? Does it enable some kind of caching?

Reply:
We are currently using the build command to cache the `agent` and `weblog (and all the variations)` as part of the `build-harness` and `build-apps`

The `./build.sh --images runner` makes sure to install all the Python dependencies in the host machine to be able to execute `./run.sh ${{ matrix.scenario }}`

Here we would benefit from sharing the Python package dependencies between the different `./run.sh` jobs. Since each job step happens on a

Reply:
That's one of the main reasons of using Docker in CI instead of running on host.

The main reason to run on host was to facilitate development, at the expense of CI. The system tests repo's own CI uses a single job that does build once, then runs all scenarios _linearly_, missing opportunities for parallelization.

I'm trying to restore the availability of previous behaviour in https://github.com/DataDog/system-tests/pull/1109, which _also_ solves reaching out to a non-localhost Docker when running on host. See https://github.com/DataDog/system-tests/pull/1095#issuecomment-1514595189.

Reply:
That sounds way better! Our CI seems to be expanding and slowing down, so all work to make it faster is makes me very very happy :)